packages/flutter/lib/src/widgets/inherited_model.dart

```diff
@@ -0,0 +1,234 @@
+  /// [InheritedModel] ancestor of type T whose [InheritedModel.aspects] set
+  /// is either null (the default) or contains [aspect]. If [aspect] is
+  /// null then it's the first ancestor of type T.
+  ///
+  /// If [aspect] is null this method is the same as
+  /// `context.inheritFromWidgetOfExactType(T)`.
+  static T inheritFrom<T extends InheritedModel<Object>>(BuildContext context, { Object aspect }) {
+    if (aspect == null)
+      return context.inheritFromWidgetOfExactType(T);
+
+    // Create a dependency on all of the type T ancestor models up until
+    // a model is found whose aspects are null or contain the given aspect.
+    // Rebuilding one of the intermediate models only causes its dependents
+    // to be rebuilt if its aspects property changes (see _shouldNotify).
+    final List<InheritedElement> models = <InheritedElement>[];
+    _findModels<T>(context, aspect, models);
+    if (models.isEmpty)
+      return null;
+    final InheritedElement lastModel = models.last;
+    for (InheritedElement model in models) {
+      final T value = context.inheritFromElement(model, aspect: aspect);
+      if (model == lastModel)
+        return value;
+    }
+    assert(false, 'Unreachable');
+    return null;
+  }
+}
+
+/// An [Element] that uses a [InheritedModel] as its configuration.
+class InheritedModelElement<T> extends InheritedElement {
+  /// Creates an element that uses the given widget as its configuration.
+  InheritedModelElement(InheritedModel<T> widget) : super(widget);
+
+  @override
+  InheritedModel<T> get widget => super.widget;
+
+  @override
+  void updateDependencies(Element dependent, Object aspect) {
+    final Set<T> dependencies = getDependencies(dependent);
+    if (dependencies != null && dependencies.isEmpty)
+      return;
+
+    if (aspect == null) {
+      setDependencies(dependent, new HashSet<T>());
+    } else {
+      assert(aspect is T);
+      setDependencies(dependent, dependencies == null
+        ? (new HashSet<T>()..add(aspect))
+        : dependencies..add(aspect));
+    }
+  }
+
+  bool _shouldNotify(InheritedModel<T> oldWidget, Element dependent, Set<T> aspects) {
+    // If the aspects the model supports have changed, then rebuild.
+    if (_notEqual<T>(widget.aspects, oldWidget.aspects))
+      return true;
+
+    // The aspects argument specifies the model aspects that the dependent element depends on.
+    if (widget.aspects == null || widget.aspects.any((T aspect) => aspects.contains(aspect)))
+      return widget.updateShouldNotifyDependent(oldWidget, aspects);
+
+    return false;
+  }
+
+  @override
+  void notifyDependent(InheritedModel<T> oldWidget, Element dependent) {
+    final Set<T> dependencies = getDependencies(dependent);
+    if (dependencies == null)
+      return;
+    if (dependencies.isEmpty || _shouldNotify(oldWidget, dependent, dependencies))
+      dependent.didChangeDependencies();
+  }
+}
+
+bool _notEqual<T>(Set<T> a, Set<T> b) {
```

Comment:
You can use setEquals from the foundation library

Reply:
Thanks, I didn't know of setEquals.